src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -8077,6 +8077,8 @@ namespace InternalHeatGains {
                 Found = true;
                 DeviceIndex = DeviceNum;
                 break;
+            } else {
+                ErrorFound = true;
```

Comment:
I guess this change works technically, but it seems out of place.  With this change, you basically have this:

```
Loop over all devices
  If device.name matches searchName
    If device.type matches searchType
      ErrorsFound = false, break, and return
    But if device.type doesn't match searchType
      ErrorsFound = true, break, and return
    EndIf
  But if device.name doesn't match searchName    << new line
    ErrorsFound = true, and continue the loop    << new line
  EndIf
EndLoop
```

So it's weird because you are going to loop over all the devices and you are going to end up setting ErrorsFound to true a good number of times before you find a match.  Why go through the trouble of setting ErrorsFound so many times?  It should've just been initialized to true before the loop starts and that's all you needed to do.  Then it would remain false if the loop terminates without finding a match.  And in fact, you could just simplify the logic and remove the line that sets ErrorsFound to false because it would already be initialized to false.  It should be reduced to something like:

```
ErrorsFound = true  << assume we don't find a match
Loop over all devices
  If device.name matches searchName
    If device.type matches searchType
      ErrorsFound = false
    EndIf
    break and return
  EndIf
EndLoop
```

Seems like a simpler solution, and probably worth making the change before this merges.  Or at least let me know why you think we shouldn't do it.

Reply:
@Myoldmopar When I first saw the comment about `ErrorFound` I was concerned that this may be a case where the argument is passed along to many functions and might already be true when this function is called (as so many other places in the code). But in this case, the argument is [localized when the function is called](https://github.com/NREL/EnergyPlus/blob/6b0e48b11868effc547dade7162791519d232536/src/EnergyPlus/RoomAirModelManager.cc#L1713-L1728). Seems it would be better for this function to be a `bool` and remove `ErrorFound` as an argument, just to keep it clear for future devs, but maybe that's nit-picking. Or maybe it should be an `int` function and keep the error arg.

Reply:
I think it would be reasonable for the function to just return an integer and eliminate the error found bool altogether.  Return -1 as an invalid device index.  That would eliminate the need to return **both** an error flag and a valid index flag.  And that is something we do quite often in E+ already.  (Not that that alone would be precedent...)

One additional note is that if you make a change to initialize the variable to true you may need to add it to the approve list in `scripts/dev/find_byref_bool_override.py`.

Reply:
@mjwitte @Myoldmopar Thank you for the comments and notes! I will update this as an int function. Should I consider moving the related error message from RoomAirModelManager.cc inside this function?

Reply:
> Should I consider moving the related error message from RoomAirModelManager.cc inside this function?

I would leave it where it's at, since it uses some local information to build the error message.

Reply:
@mjwitte @Myoldmopar  The function `GetInternalGainDeviceIndex` was updated. Please let me know for any further suggestions or comments? Thanks a lot!